### PR TITLE
Refactor functions related to file renames

### PR DESCRIPTION
**Pull request by gamemaker1**

### CODE REVIEW

**Comment by yakov116** [comment]
@gamemaker1 can you fix the merge conflict?

**Comment by gamemaker1** [comment]
Sure I'll do it now

**Comment by gamemaker1** [comment]
Done!

File: source/features/follow-file-renames.tsx [hunk 1]
```diff
@@ -8,50 +8,69 @@ import * as api from '../github-helpers/api';
...
+	const commitInfo: CommitInfo = {
+		parentSha: commit.parents[0],
+		date: commit.commit.committer.date,
+		url: commit.html_url,
+	};
+	for (const fileInfo of commit.files as File[]) {
+		if ([fileInfo.filename, fileInfo.previous_filename].includes(filePath)) {
+			return {
+				commit: commitInfo,
+				file: fileInfo,
```
Comment (fregante) [inline source/features/follow-file-renames.tsx:38]: I meant this
```suggestion
				commit: {
		 		    parentSha: commit.parents[0],
		 		    date: commit.commit.committer.date,
		 		    url: commit.html_url,
		 	    },
				file: fileInfo,
```

**Review by fregante — CHANGES_REQUESTED** [review]

File: source/features/follow-file-renames.tsx (hunk 1 above)
Comment (gamemaker1) [reply source/features/follow-file-renames.tsx]: Changed to the above, removed the `CommitInfo` interface

File: source/features/useful-not-found-page.tsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ function parseCurrentURL(): string[] {
 	return parts;
 }
 
-async function getLatestChangeToFile(): Promise<Record<string, any> | void> {
+async function getLatestChangeToFile(): Promise<FileChanges | void> {
```
[the remark this answers is not in the document]
Comment (gamemaker1) [reply source/features/useful-not-found-page.tsx]: Done

File: source/features/follow-file-renames.tsx [hunk 3]
```diff
@@ -8,50 +8,69 @@ import * as api from '../github-helpers/api';
...
-export async function getCommitInfo(oid: string): Promise<AnyObject> {
+interface CommitInfo {
+	parentSha: string | undefined;
+	date: Date;
+	url: string;
+}
+export interface FileChanges {
+	commit: CommitInfo;
+	file: File;
+}
```
[the remark this answers is not in the document]
Comment (fregante) [reply source/features/follow-file-renames.tsx]: ONE interface

File: source/features/follow-file-renames.tsx [hunk 4]
```diff
@@ -8,50 +8,67 @@ import * as api from '../github-helpers/api';
...
-				);
-			}
+	const fileChanges = await getChangesToFileInCommit(sha.getAttribute('value')!, filePath);
+	if (fileChanges?.file?.status !== 'renamed') {
+		return;
+	}
 
-			return;
-		}
+	if (fileChanges.file.status === 'renamed') {
```
Comment (fregante) [inline source/features/follow-file-renames.tsx:53]: Duplicate condition

File: source/features/follow-file-renames.tsx [hunk 5]
```diff
@@ -8,50 +8,67 @@ import * as api from '../github-helpers/api';
...
-						className="btn btn-outline BtnGroup-item tooltipped tooltipped-n tooltipped-no-delay"
-					>
-						{isNewer && <DiffRenamedIcon className="mr-1" style={{transform: 'rotate(180deg)'}}/>}
-						{button.textContent}
-						{!isNewer && <DiffRenamedIcon className="ml-1"/>}
-					</a>,
-				);
-			}
+	const fileChanges = await getChangesToFileInCommit(sha.getAttribute('value')!, filePath);
+	if (fileChanges?.file?.status !== 'renamed') {
```
Comment (fregante) [inline source/features/follow-file-renames.tsx:49]:
```suggestion
	if (fileChanges?.file.status !== 'renamed') {
```

**Review by fregante — APPROVED** [review]
Untested but I think we’re good after this

**Comment by gamemaker1** [comment]
Removed the duplicate condition, fixed another error when viewing a 404 for a renamed file - now it works! 🎉

File: source/features/follow-file-renames.tsx [hunk 6]
```diff
@@ -8,51 +8,66 @@ import * as api from '../github-helpers/api';
...
 	filename: string;
 	status: string;
+	blob_url: string;
 }
-// eslint-disable-next-line import/prefer-default-export
-export async function getCommitInfo(oid: string): Promise<AnyObject> {
+export interface FileChanges {
+	file: File;
+	commit: {
+		parentSha: string | undefined;
```
Comment (fregante) [inline source/features/follow-file-renames.tsx:19]: When would this undefined?

File: source/features/follow-file-renames.tsx [hunk 7]
```diff
@@ -8,51 +8,66 @@ import * as api from '../github-helpers/api';
 import GitHubURL from '../github-helpers/github-url';
 
 interface File {
-	previous_filename: string;
+	previous_filename: string | undefined;
```
Comment (fregante) [inline source/features/follow-file-renames.tsx:11]: I prefer
```suggestion
	previous_filename?: string;
```

File: source/features/follow-file-renames.tsx (hunk 6 above)
Comment (gamemaker1) [reply source/features/follow-file-renames.tsx:19]: Maybe if it is the first commit? But then it won't be renamed or deleted - I'll change that

File: source/features/follow-file-renames.tsx (hunk 7 above)
Comment (gamemaker1) [reply source/features/follow-file-renames.tsx:11]: Will do

**Comment by gamemaker1** [comment]
Done!

**Comment by fregante** [comment]
Tested and it works. Always include some test links in your PRs. In this case they can be copied from these 2 PRs:

- https://github.com/sindresorhus/refined-github/pull/4677
- https://github.com/sindresorhus/refined-github/pull/4646

**Pull request merged**